refactor(router): simplify imports and improve option access in RouterScroller tests

SkyZeroZx opened the pull request:
Simplifies imports and unifies option access in RouterScroller sp
ecs for cleaner and more consistent code

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

JeanMeche commented on packages/router/test/router_scroller.spec.ts:
```diff
@@ -291,3 +291,163 @@ function createRouterScroller({
+    const scroller = TestBed.inject(ROUTER_SCROLLER);
+    expect((scroller as any).options.scrollPositionRestoration).toBe('disabled');
+    expect((scroller as any).options.anchorScrolling).toBe('disabled');
+  });
```
nit: No need to resort to any for private property
```suggestion
    const scroller = TestBed.inject(ROUTER_SCROLLER);
    expect(scroller['options'].scrollPositionRestoration).toBe('disabled');
    expect(scroller['options'].options.anchorScrolling).toBe('disabled');
  });
```

SkyZeroZx replied:
Done

SkyZeroZx commented on packages/router/test/router_scroller.spec.ts:
```diff
@@ -291,3 +291,163 @@ function createRouterScroller({
+    scrollPositionRestoration,
+    anchorScrolling,
+  }: InMemoryScrollingOptions) {
+    const viewportScroller = jasmine.createSpyObj('viewportScroller', [
```
Done

atscott reviewed (changes requested):
These tests very much cover the same bit of code that is already covered by the existing tests. I don't see much value added here and it actually is more of a maintenance burden when implementation details change.

SkyZeroZx commented:
So we can just remove the "TODO" and do a minor cleanup refactor for the any casting. Without adding more unit tests

SkyZeroZx committed:
refactor(router): simplify imports and improve option access in RouterScroller tests

Simplifies imports and unifies option access in RouterScroller specs for cleaner and more consistent code

thePunderWoman commented:
This PR was merged into the repository. The changes were merged into the following branches:

- main: 08a84e03e6fd2c4eee991d7609cc5b696ff300b8
